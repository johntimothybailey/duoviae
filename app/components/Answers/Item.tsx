import React, { ReactElement } from 'react'
import { AnswersItemProp } from './props'
import { Icon, IconProps, ListItem } from '@ui-kitten/components'
import V from 'voca'
import { Answer } from '../../state/modules/answers/Models'
import { StyleProp, ViewProps } from 'react-native'

const getIconProps = (item: Answer, props: ViewProps | undefined): IconProps => {
  const styleProps: StyleProp<any> = props?.style ?? {}
  if (item.isCorrect) {
    return {
      ...props,
      name: 'checkmark-outline',
      style: {
        ...styleProps,
        tintColor: 'green'
      }
    }
  } else {
    return {
      ...props,
      name: 'close-outline',
      style: {
        ...styleProps,
        tintColor: 'red'
      }
    }
  }
}

const Component = (props: AnswersItemProp): ReactElement => {
  const item: Answer = props.item
  return (
    <ListItem
      title={V.unescapeHtml(item.question)}
      description={`You answered '${item.selection}'`}
      accessoryRight={(props) => {
        const iconProps: IconProps = {
          ...getIconProps(item, props)
        }
        return <Icon {...iconProps} />
      }}
    />
  )
}

export default Component
