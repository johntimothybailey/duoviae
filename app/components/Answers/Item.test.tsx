import React from 'react'
import { render } from '@testing-library/react-native'
import Item from './Item'
import { IsCorrectAnswer, IsIncorrectAnswer } from '../../../test/fixtures/AnswersStateMachine'
import { BootComponent } from '../../../test'

describe('Answers Item', () => {
  describe('Render', () => {
    it('is correct', () => {
      const { toJSON } = render(
        <BootComponent>
          <Item item={IsCorrectAnswer} key={1} />
        </BootComponent>
      )
      expect(toJSON()).toMatchSnapshot()
    })
    it('is incorrect', () => {
      const { toJSON } = render(
        <BootComponent>
          <Item item={IsIncorrectAnswer} key={1} />
        </BootComponent>
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })
})
