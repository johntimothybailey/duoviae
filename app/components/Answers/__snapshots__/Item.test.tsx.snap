// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers Item Render is correct 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 12,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "marginHorizontal": 8,
                },
                Object {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          The video game "Fuel" has an open world that is 5,560 square miles?
        </Text>
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#8F9BB3",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "marginHorizontal": 8,
                },
                Object {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          You answered 'True'
        </Text>
      </View>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="green"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 24,
                "marginHorizontal": 8,
                "tintColor": "green",
                "width": 24,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={4278222848}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Answers Item Render is incorrect 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 12,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "600",
                  "marginHorizontal": 8,
                },
                Object {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          The video game "Fuel" has an open world that is 5,560 square miles?
        </Text>
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#8F9BB3",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "marginHorizontal": 8,
                },
                Object {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          You answered 'False'
        </Text>
      </View>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={24}
          bbWidth={24}
          fill="red"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 24,
                "marginHorizontal": 8,
                "tintColor": "red",
                "width": 24,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={4294901760}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGGroup>
                <RNSVGRect
                  height="24"
                  matrix={
                    Array [
                      -1,
                      1.2246467991473532e-16,
                      -1.2246467991473532e-16,
                      -1,
                      24,
                      24,
                    ]
                  }
                  opacity={0}
                  width="24"
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M13.41 12l4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
